cudaaligner/src/myers_gpu.cu

```diff
@@ -0,0 +1,433 @@
+        return r;
+    }
+
+    cudaStream_t stream;
+    CGA_CU_CHECK_ERR(cudaStreamCreate(&stream));
+
+    int32_t max_sequence_length = std::max(get_size(target), get_size(query));
+    device_storage<char> sequences_d(2 * max_sequence_length, device_id);
+    device_storage<int32_t> sequence_lengths_d(2, device_id);
+
+    const int32_t n_words = (get_size(query) + word_size - 1) / word_size;
+    batched_device_matrices<myers::WordType> pv(1, n_words * (get_size(target) + 1), stream, device_id);
+    batched_device_matrices<myers::WordType> mv(1, n_words * (get_size(target) + 1), stream, device_id);
+    batched_device_matrices<int32_t> score(1, n_words * (get_size(target) + 1), stream, device_id);
+    batched_device_matrices<myers::WordType> query_patterns(1, n_words * 4, stream, device_id);
+
+    batched_device_matrices<int32_t> fullscore(1, (get_size(query) + 1) * (get_size(target) + 1), stream, device_id);
+
+    std::array<int32_t, 2> lengths = {static_cast<int32_t>(get_size(query)), static_cast<int32_t>(get_size(target))};
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(sequences_d.data(), query.data(), sizeof(char) * get_size(query), cudaMemcpyHostToDevice, stream));
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(sequences_d.data() + max_sequence_length, target.data(), sizeof(char) * get_size(target), cudaMemcpyHostToDevice, stream));
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(sequence_lengths_d.data(), lengths.data(), sizeof(int32_t) * 2, cudaMemcpyHostToDevice, stream));
+
+    myers::myers_compute_score_matrix_kernel<<<1, warp_size, 0, stream>>>(pv.get_device_interface(), mv.get_device_interface(), score.get_device_interface(), query_patterns.get_device_interface(), sequences_d.data(), sequence_lengths_d.data(), max_sequence_length, 1);
+    myers::myers_convert_to_full_score_matrix_kernel<<<1, 1, 0, stream>>>(fullscore.get_device_interface(), pv.get_device_interface(), mv.get_device_interface(), score.get_device_interface(), sequence_lengths_d.data(), 0);
+
+    matrix<int32_t> fullscore_host = fullscore.get_matrix(0, get_size(query) + 1, get_size(target) + 1, stream);
+    CGA_CU_CHECK_ERR(cudaStreamSynchronize(stream));
+    CGA_CU_CHECK_ERR(cudaStreamDestroy(stream));
+    return fullscore_host;
+}
+
+void myers_gpu(int8_t* paths_d, int32_t* path_lengths_d, int32_t max_path_length,
+               char const* sequences_d,
+               int32_t const* sequence_lengths_d,
+               int32_t max_sequence_length,
+               int32_t n_alignments,
+               batched_device_matrices<myers::WordType>& pv,
+               batched_device_matrices<myers::WordType>& mv,
+               batched_device_matrices<int32_t>& score,
+               batched_device_matrices<myers::WordType>& query_patterns,
+               cudaStream_t stream)
+{
+    constexpr int32_t warp_size = 32;
+
+    {
+        const dim3 threads(warp_size, 1, 1);
+        const dim3 blocks(1, ceiling_divide<int32_t>(n_alignments, threads.y), 1);
```

Comment:
I think I'm missing something here - 
the alignment index here seems to be `blockIdx.y` (based on how many alignments are being processed in a thread block, which currently is 1). is that correct?
but in line 250, `alignment_idx = threadIdx.y` which should always be 1, right?

Reply:
That is certainly a bug and should have read `blockIdx.y*blockDim.y+threadIdx.y`. Surprising that it passed the tests. I extended the tests to catch this and discovered another bug (AlignerGlobal::run_alignment got passed `max_query_length_` instead of `max_result_length` as 3rd argument). Fixed.

Reply:
Awesome, thank you :)!